platform/src/test/java/org/exoplatform/platform/qa/ui/platform/forum/ForumPublishActivityTestIT.java

```
@@ -0,0 +1,746 @@
+   * destination forum Input Data: Expected Outcome: - Topic is moved to
+   * destination Forum successfully - In activity stream, a comment is added into
+   * activity,message is "Topic has been moved to: $value.Where $value is
+   * :Space>Category>Forum...
+   */
+
+  @Test
+  public void test01_CheckActivityAfterMoveATopic() {
+    info("Test 01: Check activity after move a topic");
+    info("Create data test");
+    prepareDataTest();
+    String Forum = "Forum" + getRandomNumber();
+    String Cat = "Category" + getRandomNumber();
+    String Topic = "Topic" + getRandomNumber();
+    String comment = "Topic have been moved to: " + nameCat + ">" + nameForum;
+    info("Finishing creating data test");
+
+    homePagePlatform.goToForum();
+    forumHomePage.goToHomeCategory();
+    info("Create a second Category");
+    forumCategoryManagement.addCategorySimple(Cat, "", Cat);
+    info("Create a second forum");
+    forumForumManagement.addForumSimple(Forum, "", Forum);
+
+    info("Start a topic in second forum of second Category");
+    forumForumManagement.goToStartTopic();
+    forumTopicManagement.startTopic(Topic, Topic, "", "");
+
+    info("Move Topic:" + Topic);
+    forumHomePage.goToTopic(Topic);
+    forumTopicManagement.selectItemMoreActionMenuTopic(ForumTopicManagement.specifMoreActionMenuTopic.MOVE);
+    info("Move the topic to a forum");
+    forumTopicManagement.moveTopicToForum(nameCat, nameForum);
+    info("Verify that the forum is moved to new category");
+    $(byText(Forum)).should(Condition.exist);
+    $(byText(nameCat)).should(Condition.exist);
+    $(byText(Topic)).should(Condition.exist);
+    info("Verify that the topic's activity is shown on intranet");
+    homePagePlatform.goToHomePage();
+    for (int i = 0; i <= 10; i++) {
+      refresh();
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        e.printStackTrace();
```

How do you deal with exception usually ? no logger ?

---

yes,I must log it ?

---

How do you manage exceptions usually ?
Actually the best thing to do it maybe to just throw the exception (remove the try/catch and add throws Exception in the method declaration). WDYT ?

---

Yes,I change it